ShearInZ/ShearInZ.c

```diff
@@ -1,5 +1,20 @@
 #include "ShearInZ.h"
 
 void zShear() {
-    //test
+    // Use the transformation matrix and multiply the vectors 
+    struct point **currentVector = NULL;
+
+    // Reset transformation matrix before use
+    resetMatrix();
+
+    // store Shear Values
+    transformationMatrix[2][0] = getXShear();
+    transformationMatrix[2][1] = getYShear();
+    
+    // Loop call of multiply martrix on current point
+    for(int i = 0; i < inputShape->numOfPoints; i++) {
+        *currentVector = getPoint(i);

```

Comment:
the issues on the server are right and are caused by this code here (please look at reflection in Y for clarity on how to use Getpoints with multiply matrix and setPoint